src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildAnAppWithTransitiveNonSdkProjectRefs.cs

```diff
@@ -0,0 +1,192 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Xml.Linq;
+
+using FluentAssertions;
+using Microsoft.Extensions.DependencyModel;
+using Microsoft.NET.TestFramework;
+using Microsoft.NET.TestFramework.Assertions;
+using Microsoft.NET.TestFramework.Commands;
+using Microsoft.NET.TestFramework.ProjectConstruction;
+
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.NET.Build.Tests
+{
+    public class GivenThatWeWantToBuildAnAppWithTransitiveNonSdkProjectRefs : SdkTest
+    {
+        public GivenThatWeWantToBuildAnAppWithTransitiveNonSdkProjectRefs(ITestOutputHelper log) : base(log)
+        {
+        }
+
+        [WindowsOnlyFact]
+        public void It_builds_the_project_successfully()
+        {
+            // NOTE the projects created by CreateTestProject:
+            // TestApp --depends on--> MainLibrary --depends on--> AuxLibrary (non-SDK)
+            // (TestApp transitively depends on AuxLibrary)
+            var testAsset = _testAssetsManager
+                .CreateTestProject(CreateTestProject());
+
+            VerifyAppBuilds(testAsset, string.Empty);
+        }
+
+        [WindowsOnlyTheory]
+        [InlineData("")]
+        [InlineData("TestApp.")]
+        public void It_builds_deps_correctly_when_projects_do_not_get_restored(string prefix)
+        {
+            // NOTE the projects created by CreateTestProject:
+            // TestApp --depends on--> MainLibrary --depends on--> AuxLibrary
+            // (TestApp transitively depends on AuxLibrary)
+            var testAsset = _testAssetsManager
+                .CreateTestProject(CreateTestProject())
+                .WithProjectChanges(
+                    (projectName, project) =>
+                    {
+                        string projectFileName = Path.GetFileNameWithoutExtension(projectName);
+                        if (StringComparer.OrdinalIgnoreCase.Equals(projectFileName, "AuxLibrary") ||
+                            StringComparer.OrdinalIgnoreCase.Equals(projectFileName, "MainLibrary"))
+                        {
+                            var ns = project.Root.Name.Namespace;
+
+                            if (!string.IsNullOrEmpty(prefix))
+                            {
+                                XElement propertyGroup = project.Root.Element(XName.Get("PropertyGroup", ns.NamespaceName));
+                                XElement assemblyName = propertyGroup.Element(XName.Get("AssemblyName", ns.NamespaceName));
```

Comment:
I believe using the namespace can be simplified here:

```suggestion
                                XElement propertyGroup = project.Root.Element(ns + "PropertyGroup");
                                XElement assemblyName = propertyGroup.Element(ns + "AssemblyName");
```

Reply:
Yep!